### ISSUE TITLE
Sketcher: poor UI visibility of dimension labels and construction geometry with the default settings

### ISSUE THREAD

**Issue by FEA-eng** [FreeCAD/FreeCAD#13318]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

This was raised during the discussion about #10100 but I'm talking specifically about the dimension labels and construction geometry and with no relation to colorblindness. IMO their visibility is really bad with the default settings, especially within the darker (top) region of the background:

![dim](https://github.com/FreeCAD/FreeCAD/assets/59876896/11c250bc-30ce-40bb-8608-7a036962684b)

I even find it difficult to prepare good-looking screenshots for Sketcher release notes.

I would like to propose 2 solutions (that can be implemented regardless of the big issue about colorblindness):
1) changing the default dimension and construction geometry color to something more visible on the default background (white? yellow?)
2) changing the default background to a lighter one

I would probably go with 1 since 2 is a bigger decision. Here's what it looks like in Abaqus which also uses a dark background:

![dim abq](https://github.com/FreeCAD/FreeCAD/assets/59876896/00b32eff-75d2-45c0-b6df-9187ce206612)

I don't say it's perfect but at least everything is easily distinguishable. It seems that CATIA (another dark blue background) uses green for both dimensions and constrained sketch geometry but we should probably stay with separate colors for them.

### Full version info

```shell
OS: Windows 10 build 19045
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.36731 (Git)
Build type: Release
Branch: main
Hash: ac1d887336826c59321352f49a195bf1ef8be2f0
Python 3.11.8, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: Polish/Poland (pl_PL)
```

### Subproject(s) affected?

Sketcher

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by obelisk79** [comment]
I would propose that the text label of dimensions should have a small solid color background. Otherwise there will always be circumstances where readability may suffer with solid geometry displayed while sketching.

**Comment by PaddleStroke** [comment]
Default colors sucks imo. I think we should change all the colors to something like OpenLight theme colors. No point in keeping default freecad very bad.

**Comment by FEA-eng** [comment]
> Default colors sucks imo. I think we should change all the colors to something like OpenLight theme colors. No point in keeping default freecad very bad.

There’s an ongoing discussion about themes here: https://github.com/FreeCAD/FreeCAD/pull/13650

Unfortunately, it’s going in the direction of keeping modern themes as add-ons and including only the old ones in the default package. I think that 1.0 is a good opportunity to refresh the default appearance of FreeCAD.

**Comment by pierreporte** [comment]
The shadowed text isn’t a default theme issue though. I remember that it wasn’t several years ago. Now, the lighter the text, the more blurry. The screenshot shows this: red is lighter than blue (for the eyes, not the computer) thus less readable even on the darker side of the 3D view.

I don’t know of a single program that have a background for dimensions since their themes are all better (in this matter at least) and they don’t have shadowed text. Though there are sometimes cases when the background is too close to the dimension color, especially when drawing above visible geometry (looking at you Catia). It can still be a good idea but it needs to be very discreet. I propose to use the average background color and to put the rectangle in the background, but in front of everything that isn’t in the sketch. Maybe with transparency to avoid masking things too much.

The default theme indeed sucks though. This should be solved by integrating two new official themes, light and dark, that are better designed. OpenLight or OpenDark for instance but not necessarily, unless we want them in time for 1.0.

**Comment by FEA-eng** [comment]
I'll just add that it's even worse for external geometry now:

![external 1](https://github.com/FreeCAD/FreeCAD/assets/59876896/c4f1893f-03eb-42aa-82e8-25b2747933ff)
![external 2](https://github.com/FreeCAD/FreeCAD/assets/59876896/06c980ce-1f40-45af-abb2-52754296b684)

Perhaps it should be thicker by default?

**Comment by pierreporte** [comment]
I don’t think thicker lines are a solution. A different color could be better. My proposition is always to use color for constrain status only, so everything is white or green. External geometry would be an exception, with a contrasting color against both white and green, and the background. By the way, if the grid was more muted, it would be better.

**Comment by FEA-eng** [comment]
Yeah, but there’s no way we come to the conclusion regarding new default colors for the upcoming release so making the lines thicker can be a workaround until the colors are improved. And maybe then it will stay anyway. It’s really bad now so any change making it better in 1.0 makes sense IMO.

**Comment by pierreporte** [comment]
I’m going to propose something for the @FreeCAD/design-working-group soon, but it would be sad to thicken lines now only to make them thinner in the next release.

**Comment by FEA-eng** [comment]
UI freeze is tomorrow, I’m afraid that it’s either this potentially (but not necessarily) temporary solution or nothing. Currently, the visibility is really bad (external geometry is almost invisible) and it would be a shame if it stayed like this for the whole 1.0 release.

**Comment by PaddleStroke** [comment]
The solution is to change the background color imo.

**Comment by MiqCG** [comment]
> The solution is to change the background color imo.

With the Light theme and default colors, the external geometries are almost imperceptible, too. 
Revision 38419.

![Captura de pantalla_2024-08-12_12-03-55](https://github.com/user-attachments/assets/5dfc2111-2911-4da0-af63-ea60da30fa47)

**Comment by luzpaz** [comment]
Is there a proposal and current workaround? (we can then mark this as a post-1.0 ticket.)

**Comment by obelisk79** [comment]
> Is there a proposal and current workaround? (we can then mark this as a post-1.0 ticket.)

I still believe that textual fields for dimensional constraints should have a solid background which can either be defined by preference or hidden property. It strikes me as the only viable solution because sketches inherently have contrasting imagery to contend with. Either the background (gradient) or objects in view (solid/surface, etc ) simple color change of text, or Even the background would be insufficient due to the broad range of contrasting colors involved.

**Comment by maxwxyz** [comment]
isn't this the same as https://github.com/FreeCAD/FreeCAD/issues/10100 ?

**Comment by FEA-eng** [comment]
> isn't this the same as #10100 ?

No, I mentioned this in the issue description:

> This was raised during the discussion about https://github.com/FreeCAD/FreeCAD/issues/10100 but I'm talking specifically about the dimension labels and construction geometry and with no relation to colorblindness.

**Comment by maxwxyz** [comment]
With Classic (default) you cannot see the geometry when it is drawn:
![Bildschirmfoto 2024-09-03 um 17 58 07](https://github.com/user-attachments/assets/bcc978aa-b8e7-443a-b9d1-8e1d4b835cc1)

```
OS: macOS 13.6.9
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38622 (Git)
Build type: Release
Branch: main
Hash: ed97f3c4591303ee3b54906768f8f4eba5211265
Python 3.11.9, Qt 5.15.13, Coin 4.0.2, Vtk 9.2.6, OCC 7.7.2
Locale: C/Default (C)
Installed mods: 
  * freecad.gears 1.2.0
  * OpenTheme 2024.8.17
  * CfdOF 1.27.5
  * Defeaturing 1.2.2
  * Manipulator 1.5.7
  * fasteners 0.5.27
  * CurvedShapes 1.0.9
  * sheetmetal 0.4.24
  * Curves 0.6.45

```

**Comment by maxwxyz** [comment]
Is this still an issue with the updated light/dark theme @FEA-eng ?

**Comment by FEA-eng** [comment]
> Is this still an issue with the updated light/dark theme [@FEA-eng](https://github.com/FEA-eng) ?

It's much better with those themes but the question is if it shouldn't be improved in the Classic theme (still present although a bit hidden) too.

**Comment by pierreporte** [comment]
Regardless of the theme, the grid still looks absolutely awful though and it is displayed by default.

**Comment by MisterMakerNL** [comment]
We discussed this on discord and pretty much came too conclusion thus can only properly be fixed if we add add some kind of overlay between the model and sketch.
Because not only the background but the model itself causes contrast issues.
